### ISSUE TITLE
List-like behavior for FeatureCollections 

### ISSUE THREAD

**Issue by dericke** [jazzband/geojson#154]
In almost all cases in which I am using a `FeatureCollection`, I'm only interacting with its `features` member. In fact, I typically wait until the last possible moment to construct a `FeatureCollection` because it's easier to just work with lists of `Features`. It would be a convenient shorthand to alias the methods of the features member within the main FeatureCollection class, so that rather than writing things like

```python3
featurecollection1["features"].append(feature1)
featurecollection1["features"] += featurecollection2["features"]
``` 
I could just write 
```python3
featurecollection1.append(feature1) 
featurecollection1 += featurecollection2
```

I think I have time to write a PR for this, but wanted to gather feedback before doing so. Do others think this is a good idea? Are there pitfalls I'm overlooking?

**Comment by chiphum** [comment]
I had the same thought when I worked with it for the first time.

**Comment by rayrrr** [comment]
I don't know whether to accept this change to be honest. The [GeoJSON spec states](https://tools.ietf.org/html/rfc7946#section-3.3):

> A FeatureCollection object has a member with the name "features".  The value of "features" is a JSON array. Each element of the array is a Feature object...

So this change, while convenient for some, might also cause confusion for others.

An alternate approach would be to set variables to be equal to the "features" members of the FeatureCollections. Since they are Python lists and variables are by reference, this would empower you to use the list operations as you prefer. So if you had `feature1`, `featurecollection1`, and `featurecollection2` on hand, you could do something like

```
fc1 = featurecollection1["features"]
fc2 = featurecollection2["features"]
fc1.append(feature1)
fc1 += fc2
```

and your `featurecollection1["features"]` will then contain `feature1` and the elements in `featurecollection2`.

Would that be a sufficient solution?

**Comment by dericke** [comment]
I can certainly understand the concern. My goal isn't to change anything already implemented in `FeatureCollection` or its superclasses, nor to hide or abstract away the structure of a FeatureCollection. I did mistakenly override the inherited `dict.clear()` in the related PR, which is now corrected. I was under the impression that implementing otherwise-unimplemented methods and operators would be relatively safe.

Regardless of whether this change is accepted, I appreciate the workaround you've offered, and will probably use that in the future.
